### PR TITLE
Move endpoints IPPart function to util

### LINKED ISSUE #53632: Move endpoints IPPart() to util
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind cleanup

**What happened**:

The latest IPPart() supports IPV6, we need to update it in proxier.go

**What you expected to happen**:

Move endpoints IPPart function to util so that it can be consumed by different proxiers.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig network
/area kube-proxy
/area ipv6
/assign @m1093782566

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Clean up proxier.go

Move endpoints IPPart function to util so that it can be consumed by different proxiers.

BTW, the newer `IPPart()` supports IPV6, we need to update it in `ipvs/proxier.go`

**Which issue this PR fixes**:

 closes #53632

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig network
/area kube-proxy

### CODE REVIEW

File: pkg/proxy/util/endpoints.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		glog.Errorf("Error parsing '%s': %v", s, err)
+		return ""
+	}
+	return ip
+}
+
+// HostAddress returns a host address of the form <ip-address>/32 for
+// IPv4 and <ip-address>/128 for IPv6
+func HostAddress(ip net.IP) string {
```
Comment (thockin) [inline pkg/proxy/util/endpoints.go]: Can we rename this to `HostCIDR` or `AsCIDR` or `ToCIDR` ?

Comment (m1093782566) [reply pkg/proxy/util/endpoints.go]: I feel like `ToCIDR` is a good name...

**Pull request merged**